backend/main/server/handlers/communities.go

```diff
@@ -0,0 +1,238 @@
+		if err != nil {
+			s.RespondWithError(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+		response := s.GetPaginatedResponseWithPayload(communities, start, count, totalRecords)
+	
+		s.RespondWithJSON(w, http.StatusOK, response)
+	}
+}
+
+func getCommunity(db *s.Database) func (w http.ResponseWriter, r *http.Request) {
+	return func (w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		id, err := strconv.Atoi(vars["id"])
+
+		if err != nil {
+			s.RespondWithError(w, http.StatusBadRequest, "Invalid Community ID.")
+			return
+		}
+
+		c := models.Community{ID: id}
+		if err := c.GetCommunity(db); err != nil {
+			// TODO: for some reason switch err doesn't match pgx.ErrNoRows.
+			// So I've added .Error() to convert to a string comparison
+			switch err.Error() {
+			case pgx.ErrNoRows.Error():
+				s.RespondWithError(w, http.StatusNotFound, "Community not found.")
+			default:
+				s.RespondWithError(w, http.StatusInternalServerError, err.Error())
+			}
+			return
+		}
+
+		s.RespondWithJSON(w, http.StatusOK, c)
+	}
+}
+
+func getCommunitiesForHomePage(db *s.Database) func (w http.ResponseWriter, r *http.Request) {
+	return func (w http.ResponseWriter, r *http.Request) {
+		count, _ := strconv.Atoi(r.FormValue("count"))
+		start, _ := strconv.Atoi(r.FormValue("start"))
+
+		if count > 25 || count < 1 {
+			count = 25
+		}
+		if start < 0 {
+			start = 0
+		}
+
+		communities, totalRecords, err := models.GetCommunitiesForHomePage(db, start, count)
+		if err != nil {
+			s.RespondWithError(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+		response := s.GetPaginatedResponseWithPayload(communities, start, count, totalRecords)
+
+		s.RespondWithJSON(w, http.StatusOK, response)
+	}
+}
+
+func createCommunity(db *s.Database, ipfsClient *s.IpfsClient, flowAdapter *s.FlowAdapter, config s.Config) func (w http.ResponseWriter, r *http.Request) {
+	return func (w http.ResponseWriter, r *http.Request) {
+		var c models.Community
+		var payload models.CreateCommunityRequestPayload
+
+		decoder := json.NewDecoder(r.Body)
+
+		if err := decoder.Decode(&payload); err != nil {
+			log.Error().Err(err)
+			s.RespondWithError(w, http.StatusBadRequest, "Invalid request payload.")
+			return
+		}
+		defer r.Body.Close()
+
+		c = payload.Community
+
+		if err := validateTimestamp(c.Timestamp, 60, config); err != nil {
+			s.RespondWithError(w, http.StatusForbidden, err.Error())
+			return
+		}
+
+		// TODO: since we removed the allowlist, some sort of rate limiting will probably be necessary
+		if err := validateSignature(c.Creator_addr, c.Timestamp, c.Composite_signatures, flowAdapter, config); err != nil {
+			s.RespondWithError(w, http.StatusForbidden, err.Error())
+			return
+		}
+
+		pin, err := ipfsClient.PinJson(c)
+		if err != nil {
+			s.RespondWithError(w, http.StatusInternalServerError, "IPFS error: "+err.Error()+".")
+			return
+		}
+		c.Cid = &pin.IpfsHash
+
```

Comment:
ipfs Pin logic also repeats throughout all the controllers,

Reply:
Created helper function, although it has minimal impact on code length.